packages/core/src/new-api/types/deployer.ts

```diff
@@ -22,6 +23,17 @@ export interface DeploymentResultSuccess {
   module: IgnitionModule<string, string, IgnitionModuleResult<string>>;
 }
 
+/**
+ * The result of a failed deployment run (at least one future had an
+ * failed transaction).
+ *
+ * @beta
+ */
+export interface DeploymentResultFailure {
+  status: "failure";
+  errors: { [key: string]: Error };
```

Comment:
In a similar fashion than https://github.com/NomicFoundation/ignition/pull/284/files#r1244391873, I don't think Error is the right abstraction here. We should have a more precise type, possibly a union of different possible errors.